src/common.h

```diff
@@ -76,6 +76,7 @@
 
 #include "git2/types.h"
 #include "git2/errors.h"
+#include "git2/deprecated.h"
```

Comment:
Shouldn't this have a comment as to why it's there?  It feels like it'd be easy for somebody to think this is useless and drop it.  Especially if you don't include your deprecated functions test case, but even then they might loose some of their time.

Reply:
🤔   Seems reasonable enough.